File: src/2490.circular-sentence.2580/solution.js
/**
 * @param {string} sentence
 * @return {boolean}
 */
var isCircularSentence = function (sentence) {
    let index = 0;
    while (index < sentence.length) {
        if (sentence[index] !== ' ') {
            index++;
            continue;
        }
        if (sentence[index - 1] !== sentence[index + 1]) {
            return false;
        }
        index++;
    }
    return sentence[0] === sentence[sentence.length - 1];
};

File: src/2490.circular-sentence.2580/solution.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        String[] strings = sentence.split(" ");

        for(int i=0;i< strings.length;i++){
            String current = strings[i];
            String next = strings[(i+1)%strings.length];
            if(next.charAt(0) != current.charAt(current.length()-1)){
                return false;
            }
        }
        return true;
    }
}